src/NHibernate/Linq/Visitors/ExpressionParameterVisitor.cs

```diff
@@ -107,6 +102,18 @@ protected override Expression VisitMethodCall(MethodCallExpression expression)
 				return Expression.Call(null, expression.Method, query, arg);
 			}
 
+			if (_functionRegistry != null && 
```

Comment:
While we do not have other functions that accept collection arguments it would be nice if we can generalize this logic. I was trying to implement support for `string.Concat(...)` which has an overload with `params`. I had stuck on the over-caching issue for different parameter types.

Reply:
I agree and added `TryGetCollectionParameter` method to `IHqlGeneratorForMethodExtended` interface. Unfortunately the method name check optimization is now gone, but given that `DefaultLinqToHqlGeneratorsRegistry.TryGetGenerator` caches the result, I think it should be fine.